### PR TITLE
Add TryFrom for str type for PrintableString

### PR DESCRIPTION
**Pull request by Nicceboy**
Also small documentation updates, enforce use of `from` or `try_from`, same trait order

### CODE REVIEW

File: src/types/strings/printable.rs [hunk 1]
```diff
@@ -15,21 +20,25 @@ impl StaticPermittedAlphabet for PrintableString {
...
-            "{} not in character set",
-            ch
+            "{ch} not in character set"
         );
         self.0.push(ch as u8);
     }
+
+    fn chars(&self) -> Box<dyn Iterator<Item = u32> + '_> {
+        Box::from(self.0.iter().map(|byte| *byte as u32))
+    }
```
Comment (XAMPPRocky) [inline src/types/strings/printable.rs:33]: Why `u32` when it can fit in a `u8`? Also why use `Box<dyn Iterator>` instead of `impl Iterator`?

Comment (Nicceboy) [reply src/types/strings/printable.rs:33]: Those were already there, I just changed the order to be same as in Trait. I have no idea 😅 

https://github.com/XAMPPRocky/rasn/blob/a7a00c76d702e530a14d517d830adc899bd31dc4/src/types/strings/printable.rs#L18-L20

I can change those at the same time.

Comment (XAMPPRocky) [reply src/types/strings/printable.rs:33]: Oh I didn't notice this was in the trait, in that case it makes sense because you can't use either of those.

**Comment by XAMPPRocky** [comment]
Thank you for your PR!

**Pull request merged**